src/test/compile-fail/call-fn-never-arg-wrong-type.rs

```diff
@@ -8,8 +8,13 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-fn forever2() -> ! { //~ ERROR computation may converge in a function marked as diverging
-  loop { break }
+#![feature(never_type)]
+
```

Comment:
These tests are great. My one nit: I'd prefer to see comments at the top explaining what precisely the test is trying to test. I find that so helpful as a reader...